pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -97,5 +116,189 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
+			} else {
+				// Automatic cluster selection
+				members := cluster.GetMemberClusters()
+				if len(members) > 0 {
+					targetCluster = members[0].Name
+				} else {
+					err := r.setStatusNoClustersAvailable(instance, "No member clusters found")
+					if err != nil {
+						reqLogger.Error(err, "Error updating UserSignup Status", "UserID", instance.Spec.UserID)
+						return reconcile.Result{}, err
+					}
+
+					err = NewUserSignupError("No target clusters available")
+					reqLogger.Error(err, "No member clusters found")
+					return reconcile.Result{}, err
+				}
+			}
+
+			// Set the "Provisioning" status condition
+			err := r.setStatusProvisioning(instance, "Provisioning User")
+			if err != nil {
+				reqLogger.Error(err, "Error updating UserSignup Status", "UserID", instance.Spec.UserID)
+				return reconcile.Result{}, err
+			}
+
+			// Provision the MasterUserRecord
+			userAccounts := []toolchainv1alpha1.UserAccountEmbedded{
+				{
+					TargetCluster: targetCluster,
+					Spec: toolchainv1alpha1.UserAccountSpec{
+						NSTemplateSet: toolchainv1alpha1.NSTemplateSetSpec{
+							Namespaces: []toolchainv1alpha1.Namespace{},
+						},
+					},
+				},
+			}
+
+			mur := &toolchainv1alpha1.MasterUserRecord{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: instance.Spec.UserID,
+					Namespace: config.GetOperatorNamespace(),
+				},
+				Spec: toolchainv1alpha1.MasterUserRecordSpec{
+					UserID: instance.Spec.UserID,
+					UserAccounts: userAccounts,
+				},
+			}
+
+			err = r.client.Create(context.TODO(), mur)
+			if err != nil {
+				reqLogger.Error(err, "Error creating MasterUserRecord", "UserID", instance.Spec.UserID)
+				err := r.setStatusFailedToCreateMUR(instance, "Failed to create MasterUserRecord")
+				if err != nil {
+					reqLogger.Error(err, "Error setting MUR failed status")
+					return reconcile.Result{}, err
+				}
+
+				// Error creating the MasterUserRecord - requeue the request.
+				reqLogger.Error(err, "Error creating MasterUserRecord", "UserID", instance.Spec.UserID, "TargetCluster", targetCluster)
+				return reconcile.Result{}, err
+			}
+			reqLogger.Info("Created MasterUserRecord", "UserID", instance.Spec.UserID, "TargetCluster", targetCluster)
+
+			// Update the UserSignup status conditions: set PendingApproval to false, Provisioning to true
+			updatedConditions, updated := commonCondition.AddOrUpdateStatusConditions(instance.Status.Conditions, toolchainv1alpha1.Condition{
+				Type: toolchainv1alpha1.UserSignupPendingApproval,
+				Message: "",
+				Status: corev1.ConditionFalse,
+				Reason: "Approved",
+			},
+				toolchainv1alpha1.Condition{
+					Type: toolchainv1alpha1.UserSignupProvisioning,
+					Message: "",
+					Status: corev1.ConditionTrue,
+					Reason: "Provisioning",
+				})
+
+			// If the condition values were updated, post the changes
+			if updated {
+				instance.Status.Conditions = updatedConditions
+				err = r.client.Update(context.TODO(), instance)
```

Comment:
you should rather use `r.client.Status().Update` to not trigger another reconcile.

Reply:
I wasn't aware of this, thanks for letting me know. :)  Fixed.